lib/vault/client.rb

```diff
@@ -92,7 +101,8 @@ def pool
       @lock.synchronize do
         return @nhp if @nhp
 
-        @nhp = PersistentHTTP.new("vault-ruby", nil, pool_size, pool_timeout)
+        @nhp = Net::HTTP::Persistent.new(name: "vault-ruby", pool_size:)
+        @nhp.pool.instance_variable_set(:@timeout, pool_timeout)
```

Comment:
This is definitely a little Hacky but but reasonable given that it doesn't appear there's any upstream method to set this.